source/common/network/listener_impl.h

```diff
@@ -8,6 +8,10 @@
 #include "common/event/dispatcher_impl.h"
 #include "common/event/libevent.h"
 
+#include "server/connection_handler.h"
```

Comment:
The goal is to not cross streams here. Basically server depends on common but not the other way around. (No common code should include server code).

I would make an interface for the functionality you need and then use that from the server code.